compiler/rustc_builtin_macros/src/deriving/generic/mod.rs

```diff
@@ -1532,11 +1579,76 @@ impl<'a> TraitDef<'a> {
                 let mut exprs: Vec<_> = mk_exprs(i, struct_field, sp);
                 let self_expr = exprs.remove(0);
                 let other_selflike_exprs = exprs;
+                let mut skipped_derives = SkippedDerives::None;
+                let skip_enabled = cx.ecfg.features.derive_skip
+                    || struct_field.span.allows_unstable(sym::derive_skip);
+                for attr in attr::filter_by_name(&struct_field.attrs, sym::skip) {
+                    if !skip_enabled {
+                        rustc_session::parse::feature_err(
+                            &cx.sess,
+                            sym::derive_skip,
+                            attr.span,
+                            "the `#[skip]` attribute is experimental",
+                        )
+                        .emit();
+                    }
+                    let Some(skip_attr) = ast::Attribute::meta_kind(attr) else {
+                        unreachable!()
+                    };
```

Comment:
That's just `.unwrap()`:

```suggestion
                    let skip_attr = ast::Attribute::meta_kind(attr).unwrap();
```